### PR TITLE
feat: downstream mTLS

### PR DESCRIPTION
**Pull request by arkodg**
Relates to https://github.com/envoyproxy/gateway/pull/2483

### CODE REVIEW

File: internal/xds/translator/listener.go [hunk 1]
```diff
@@ -400,6 +399,16 @@ func buildDownstreamQUICTransportSocket(tlsConfig *ir.TLSConfig) (*corev3.Transp
 			})
 	}
 
+	if tlsConfig.CACertificate != nil {
```
Comment (guydc) [inline internal/xds/translator/listener.go:402]: consider also sending the client certificate upstream, e.g.  by updating the HCM default for [forwarding](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#enum-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-forwardclientcertdetails) to `SANITIZE_SET`. This is not strictly required for the use case in #88, but is a common requirement when a proxy performs client cert auth.

Comment (arkodg) [reply internal/xds/translator/listener.go:402]: I'll raise a separate issue for propagating `XFCC`

File: internal/gatewayapi/clienttrafficpolicy.go [hunk 2]
```diff
@@ -490,4 +500,55 @@ func translateListenerTLSParameters(tlsParams *egv1a1.TLSSettings, httpIR *ir.HT
...
+					return err
+				}
+
+				secretBytes, ok := secret.Data[caCertKey]
+				if !ok || len(secretBytes) == 0 {
+					return fmt.Errorf(
+						"caCertificateRef not found in secret %s", caCertRef.Name)
+				}
+
+				irCACert.Certificate = append(irCACert.Certificate, secretBytes...)
```
Comment (guydc) [inline internal/gatewayapi/clienttrafficpolicy.go:528]: should we also verify that this is an x509 conforming certificate? What's the impact of a malformed certificate provided via SDS on translation of other resources/data plane behavior?

Comment (arkodg) [reply internal/gatewayapi/clienttrafficpolicy.go:528]: raised https://github.com/envoyproxy/gateway/issues/2601 to track this

**Review by guydc — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Pull request merged**